assembly/net/index.ts

```diff
@@ -244,30 +401,50 @@ export class TCPSocket {
       : 0;
   }
 
+  /** Flush the socket. Returns true if the operation was successful. */
   public flush(): bool {
     return tcp_flush(this.socket_id) == TCPFlushResult.Success;
   }
 
+  /**
+   * This method dereferences a socket. It does not explicitly close the socket.
+   * Once a socket has been dereferenced on every Process it has been sent to,
+   * it will be closed by lunatic internally.
```

Comment:
I think it would be clearer here (and bellow for the Server `close()` method) if we explained it as: "TCP streams are reference counted. Every time a TCP stream is sent through a channel to another process it's duplicated and the reference count incremented. On every close, we decrement the the reference count. Once this reference count reaches 0, the stream is closed."

Maybe we could also rename the method to `drop()` or something similar to not confuse users that expect immediate closing of the stream.

Reply:
Sounds good to me!